libraries/core-react/src/Banner/index.ts

```diff
@@ -1,18 +1,18 @@
-import { Banner as BaseBanner } from './Banner'
+import { Banner as BaseBanner, BannerProps } from './Banner'
 import { BannerIcon } from './BannerIcon'
 import { BannerMessage } from './BannerMessage'
 import { BannerActions } from './BannerActions'
 
-type BannerProps = typeof BaseBanner & {
+type AllBannerProps = typeof BaseBanner & {
   BannerIcon: typeof BannerIcon
   BannerMessage: typeof BannerMessage
   BannerActions: typeof BannerActions
 }
 
-const Banner = BaseBanner as BannerProps
+const Banner = BaseBanner as AllBannerProps
 
 Banner.BannerIcon = BannerIcon
 Banner.BannerMessage = BannerMessage
 Banner.BannerActions = BannerActions
 
-export { Banner }
+export { Banner, BannerProps }
```

Comment:
Shouldn’t the `AllBannerProps` type be exported here? But as `BannerProps`? 🤔 So maybe import `BannerProps as BaseBannerProps`, and then rename `AllBannerProps` to `BannerProps`? Or am I wrong?

Reply:
Hard to know with these compound components… But the components has children, and those children have to be typed, and then I assume that compound type is the type for the component. This is a bit easier to grasp when this is done in the actual component and not when re-exporting. 

```tsx
type ComponentProps = {
  /** Some other colour */
  color?: string
  /** Some foo */
  foo?: string
  /** @ignore */
  children: React.ReactNode
}

type CompoundComponent = React.ForwardRefExoticComponent<ComponentProps> & {
  OtherComponent: typeof OtherComponent
}

/**
 * This is the Component description
 */

const Component = forwardRef<HTMLDivElement, ComponentProps>(function Component(
  { children, color = 'violet' },
  ref,
) {
  return (
    <div style={{ color }} ref={ref}>
      {children}
    </div>
  )
}) as CompoundComponent

Component.OtherComponent = OtherComponent
```

…except the props disappeared from the Storybook when doing it that way…

Reply:
The AllBannerProps was just a creative rename since it previously was named BannerProps 😅

Reply:
But yes, we need to figure out what is the best way of reexporting both types and props with compound components